packages/webdriverio/src/find-strategy.js

```diff
@@ -0,0 +1,183 @@
+        return 'xpath extended'
+    }
+}
+export const findStrategy = function (selector, isW3C, isMobile) {
+    /**
+     * set default strategy
+     */
+    let using = DEFAULT_STRATEGY
+    let value = selector
+
+    switch (defineStrategy(selector)) {
+    // user has specified locator strategy directly
+    case 'directly': {
+        const match = selector.match(DIRECT_SELECTOR_REGEXP)
+        if (!isMobile && isW3C && !W3C_SELECTOR_STRATEGIES.includes(match[1])) {
+            throw new Error('InvalidSelectorStrategy') // ToDo: move error to wdio-error package
+        }
+        using = match[1]
+        value = match[2]
+        break
+    }
+    case 'xpath': {
+        using = 'xpath'
+        break
+    }
+    case 'link text': {
+        using = 'link text'
+        value = selector.slice(1)
+        break
+    }
+    case 'partial link text': {
+        using = 'partial link text'
+        value = selector.slice(2)
+        break
+    }
+    case '-android uiautomator': {
+        using = '-android uiautomator'
+        value = selector.slice(8)
+        break
+    }
+    case '-ios uiautomation': {
+        using = '-ios uiautomation'
+        value = selector.slice(4)
+        break
+    }
+    case 'accessibility id': {
+        using = 'accessibility id'
+        value = selector.slice(1)
+        break
+    }
+    case 'class name': {
+        using = 'class name'
+        break
+    }
+    case 'tag name': {
+        using = 'tag name'
+        value = selector.replace(/<|>|\/|\s/g, '')
+        break
+    }
+    case 'name': {
+        if (isMobile || !isW3C) {
+            using = 'name'
+            value = selector.match(/^\[name=("|')([a-zA-z0-9\-_. ]+)("|')]$/)[2]
+        }
+        break
+    }
+    case 'xpath extended': {
+        using = 'xpath'
+        const match = selector.match(new RegExp(XPATH_SELECTOR_REGEXP.map(rx => rx.source).join('')))
+        const PREFIX_NAME = { '.': 'class', '#': 'id' }
+        const conditions = []
+        const [
+            tag,
+            prefix, name,
+            attrName, attrValue,
+            partial, query
+        ] = match.slice(1)
+
+        if (prefix) {
+            conditions.push(`contains(@${PREFIX_NAME[prefix]}, "${name}")`)
+        }
+        if (attrName) {
+            conditions.push(
+                attrValue
+                    ? `contains(@${attrName}, "${attrValue}")`
+                    : `@${attrName}`
+            )
+        }
+        if (partial) {
+            conditions.push(`contains(., "${query}")`)
+        } else {
+            conditions.push(`normalize-space() = "${query}"`)
+        }
+        value = `.//${tag || '*'}[${conditions.join(' and ')}]`
+        break
+    }
+    }
+
+    /**
+     * ensure selector strategy is supported
+     */
+    if (!isMobile && isW3C && !W3C_SELECTOR_STRATEGIES.includes(using)) {
+        throw new Error('InvalidSelectorStrategy') // ToDo: move error to wdio-error package
```

Comment:
are you going to move this within this PR?

Reply:
no, I don't want to add extra mess in this PR